[Wave 6: Referral Program] Promote referral program messaging in NewDot Redesign the Referral page

rezkiy37 opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->
The PR redesigns a page for Referral Program. Currently, it has 5 variants. For 2 (_refer a friend, share code_) of them, we changed content.

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/32187
PROPOSAL: N/A


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

#### Money request
1. Click on the **FAB**.
3. Click on **"Request money"**
4. Enter any amount.
5. Click on **"Next"**.
6. Click on **"Request money, get $250."**.
7. Verify that the page looks correctly.

#### Start chat
1. Click on the **FAB**.
2. Click on **"Start a chat"**
5. Click on **"Start a chat, get $250.**.
6. Verify that the page looks correctly.

#### Send money
1. Click on the **FAB**.
2. Click on **"Send money"**
3. Enter any amount.
4. Click on **"Next"**.
5. Click on **"Send money, get $250."**.
6. Verify that the page looks correctly.

#### Refer a friend
1. Click on the **Search**.
2. Click on **"Refer a friend, get $250."**.
3. Verify that the page looks correctly.
4. Click on **"Copy referral link"**.
5. Verify that the button label has been changed to "**Copied!**" ("**¡Copiado!**") for a moment.
6. Verify that you actually copied a link - `https://join.my.expensify.com/?thanks=<YOUR_LOGIN>`.

#### Share code
1. Go to **Settings**.
2. Click on **"Share code"**
3. Click on **"Get $250"**.
5. Verify that the page looks correctly.
6. Click on **"Copy referral link"**.
8. Verify that the button label has been changed to "**Copied!**" ("**¡Copiado!**") for a moment.
9. Verify that you actually copied a link - `https://join.my.expensify.com/?thanks=<YOUR_LOGIN>`.

#### Same for all cases:

##### "**Learn more**" click
1. Click on "**Learn more**" button on a page.
2. Verify that it redirects you to `https://help.expensify.com/articles/new-expensify/billing-and-plan-types/Referral-Program`.

##### "**Learn more**" long press/right-click with mouse
1. Long press (_on mobile devices_) / right-click with mouse (_on PC_) on "**Learn more**" button on a page.
2. Verify that a context menu with the link appears.

- [X] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->
Same as "**Tests**".

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
3. Upload an image via copy paste
11. Verify a modal appears displaying a preview of that image
--->
Same as "**Tests**".

- [X] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->
<img width="461" alt="Android Eng Learn more" src="https://github.com/Expensify/App/assets/57314631/9a98daee-1459-4572-a59e-841ad174bced">
<img width="461" alt="Android Eng Refer a friend" src="https://github.com/Expensify/App/assets/57314631/7f026fdf-2ef7-40ba-b9ac-4fc2568cf562">
<img width="461" alt="Android Eng Request money" src="https://github.com/Expensify/App/assets/57314631/94dc267f-bb2a-4f42-b341-ad6a3f9235af">
<img width="461" alt="Android Eng Send money" src="https://github.com/Expensify/App/assets/57314631/ab959432-cc34-4067-b7f7-67e5dd5b22dd">
<img width="461" alt="Android Eng Share code" src="https://github.com/Expensify/App/assets/57314631/0f5c6d23-3de8-49d7-b534-15e1f7e19a35">
<img width="461" alt="Android Eng Start a chat" src="https://github.com/Expensify/App/assets/57314631/6091c8a9-766a-433e-a084-1425752efeed">
<img width="461" alt="Android ES Learn more" src="https://github.com/Expensify/App/assets/57314631/b87855c8-e6a2-4cf5-bb0b-d0f7d2c835eb">
<img width="461" alt="Android ES Refer a friend" src="https://github.com/Expensify/App/assets/57314631/a3006a8b-04ac-4a31-8e2a-acb5effa2eee">
<img width="461" alt="Android ES Request money" src="https://github.com/Expensify/App/assets/57314631/1b8d3714-f787-4cde-8a28-4e19281c16a0">
<img width="461" alt="Android ES Send money" src="https://github.com/Expensify/App/assets/57314631/068676f2-ffa1-4eb8-b136-4dd095570543">
<img width="461" alt="Android ES Share code" src="https://github.com/Expensify/App/assets/57314631/4ebb2164-b990-45ff-9714-bbb2ddb6836f">
<img width="461" alt="Android ES Start a chat" src="https://github.com/Expensify/App/assets/57314631/19487a4b-5091-4009-b8f5-20d203576126">

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->
<img width="461" alt="Android Chrome Eng Learn more" src="https://github.com/Expensify/App/assets/57314631/29da5f6b-c9d7-48d7-ade6-88b98c7c17d5">
<img width="461" alt="Android Chrome Eng Refer a friend" src="https://github.com/Expensify/App/assets/57314631/8ecbb3dc-2a74-4fc9-8ac8-14bfc28e0dca">
<img width="461" alt="Android Chrome Eng Request money" src="https://github.com/Expensify/App/assets/57314631/01cf391b-72a8-4b45-a864-a87f6798d582">
<img width="461" alt="Android Chrome Eng Send money" src="https://github.com/Expensify/App/assets/57314631/06ba8d10-ce71-44df-9d3d-b30a92dedb3e">
<img width="461" alt="Android Chrome Eng Share code" src="https://github.com/Expensify/App/assets/57314631/f09874d4-0c68-4204-a9c1-5a30ea580018">
<img width="461" alt="Android Chrome Eng Start a chat" src="https://github.com/Expensify/App/assets/57314631/ae54baf1-85b2-416c-979d-95e055922d6a">
<img width="461" alt="Android Chrome ES Learn more" src="https://github.com/Expensify/App/assets/57314631/f9b10be6-1128-427f-b782-e0d4c07eb9ac">
<img width="461" alt="Android Chrome ES Refer a friend" src="https://github.com/Expensify/App/assets/57314631/01ff8c6e-c77d-44fe-b61b-4884a110562e">
<img width="461" alt="Android Chrome ES Request money" src="https://github.com/Expensify/App/assets/57314631/0c54242d-209c-4147-975a-f2de88f29357">
<img width="461" alt="Android Chrome ES Send money" src="https://github.com/Expensify/App/assets/57314631/1ccdf5e0-299d-4675-8c82-34da5dfd0f6d">
<img width="461" alt="Android Chrome ES Share code" src="https://github.com/Expensify/App/assets/57314631/fb5be852-01e8-48fb-966f-26efda0dfe55">
<img width="461" alt="Android Chrome ES Start a chat" src="https://github.com/Expensify/App/assets/57314631/0986693a-3d74-42d8-b1d9-b2f67f2ff667">

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->
<img width="461" alt="IOS Eng Learn more" src="https://github.com/Expensify/App/assets/57314631/94c1e81d-144f-4262-8c07-d87316780878">
<img width="461" alt="IOS Eng Refer a friend" src="https://github.com/Expensify/App/assets/57314631/780046c5-7583-4610-8f84-8a1bab1aab45">
<img width="461" alt="IOS Eng Request money" src="https://github.com/Expensify/App/assets/57314631/a8385b98-b5f1-4a4c-be82-526f025084cd">
<img width="461" alt="IOS Eng Send money" src="https://github.com/Expensify/App/assets/57314631/7de18748-8197-42d0-b020-0c6e82b69087">
<img width="461" alt="IOS Eng Share code" src="https://github.com/Expensify/App/assets/57314631/2207a54c-e65f-4985-8aca-5379dfd6bf58">
<img width="461" alt="IOS Eng Start a chat" src="https://github.com/Expensify/App/assets/57314631/e998fce4-5008-49bf-982d-fbd9e9df9bff">
<img width="461" alt="IOS ES Learn more" src="https://github.com/Expensify/App/assets/57314631/121d87fa-4d38-45e6-9efd-7331b7c04192">
<img width="461" alt="IOS ES Refer a friend" src="https://github.com/Expensify/App/assets/57314631/7d89d59d-7368-4bed-b01a-143cd4779333">
<img width="461" alt="IOS ES Request money" src="https://github.com/Expensify/App/assets/57314631/e29f0e80-65fd-4d0f-adeb-5d30b85c36e3">
<img width="461" alt="IOS ES Send money" src="https://github.com/Expensify/App/assets/57314631/dab54332-ffbd-49c9-a01f-8a7fc47ca6db">
<img width="461" alt="IOS ES Share code" src="https://github.com/Expensify/App/assets/57314631/37b7e76b-2c89-4f74-96d6-2eb7265233f7">
<img width="461" alt="IOS ES Start a chat" src="https://github.com/Expensify/App/assets/57314631/1d3c8c3b-93e1-4dba-a214-dc3906bfc382">

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->
<img width="461" alt="IOS Safari Eng Learn more" src="https://github.com/Expensify/App/assets/57314631/605287ee-bbb5-4a18-8b70-95aa303e6217">
<img width="461" alt="IOS Safari Eng Refer a friend" src="https://github.com/Expensify/App/assets/57314631/e39d36db-55dc-4127-b94c-a270ca62737b">
<img width="461" alt="IOS Safari Eng Request money" src="https://github.com/Expensify/App/assets/57314631/7be58b0d-c61f-4c68-873b-91bd7cabe105">
<img width="461" alt="IOS Safari Eng Send money" src="https://github.com/Expensify/App/assets/57314631/aa7b64e7-10d2-4e78-a870-9fff7405e569">
<img width="461" alt="IOS Safari Eng Share code" src="https://github.com/Expensify/App/assets/57314631/f573c2be-69d3-4ac1-aa45-704a9eb61837">
<img width="461" alt="IOS Safari Eng Start a chat" src="https://github.com/Expensify/App/assets/57314631/55e260c3-f302-4911-b982-f489187abcc9">
<img width="461" alt="IOS Safari ES Learn more" src="https://github.com/Expensify/App/assets/57314631/a5d02506-8f4c-4a10-868b-5280bc6e9307">
<img width="461" alt="IOS Safari ES Refer a friend" src="https://github.com/Expensify/App/assets/57314631/adba025e-a5a9-47ec-b0ce-ec6383d477b4">
<img width="461" alt="IOS Safari ES Request money" src="https://github.com/Expensify/App/assets/57314631/de4fe748-25fc-4dc4-bd41-02ce47426383">
<img width="461" alt="IOS Safari ES Send money" src="https://github.com/Expensify/App/assets/57314631/a167d1d7-2e2a-4848-a4e9-6d3b05182e42">
<img width="461" alt="IOS Safari ES Share code" src="https://github.com/Expensify/App/assets/57314631/e934d307-ca17-4a8d-b2f0-ca1e71d21f38">
<img width="461" alt="IOS Safari ES Start a chat" src="https://github.com/Expensify/App/assets/57314631/9c82d7a5-11fe-4e7f-b103-3966d3fd9bae">

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->
<img width="1728" alt="Chrome Eng Learn more" src="https://github.com/Expensify/App/assets/57314631/5715cc4a-59d4-487c-aaa8-ca30026bda9a">
<img width="1728" alt="Chrome Eng Refer a friend" src="https://github.com/Expensify/App/assets/57314631/d808f64d-0071-4f1f-a13e-77eab1954463">
<img width="1728" alt="Chrome Eng Request money" src="https://github.com/Expensify/App/assets/57314631/69a6c82c-3e8f-4a60-b734-57a24f12164a">
<img width="1728" alt="Chrome Eng Send money" src="https://github.com/Expensify/App/assets/57314631/fd3beb34-3a97-474c-a36e-00fae3836ad4">
<img width="1728" alt="Chrome Eng Share code" src="https://github.com/Expensify/App/assets/57314631/1b77de0f-a2d5-4b8d-9dc6-17a0d094ca1a">
<img width="1728" alt="Chrome Eng Start a chat" src="https://github.com/Expensify/App/assets/57314631/b591824f-889f-4a10-bdb8-948d6457ac3c">
<img width="1728" alt="Chrome ES Learn more" src="https://github.com/Expensify/App/assets/57314631/8b2f69b8-8a20-42df-95a2-2cfaa684eebb">
<img width="1728" alt="Chrome ES Refer a friend" src="https://github.com/Expensify/App/assets/57314631/6ebec22e-90a5-42d3-ad56-6e9f2c2617d7">
<img width="1728" alt="Chrome ES Request money" src="https://github.com/Expensify/App/assets/57314631/f4cf8171-41b4-4b49-b561-d4ae95526ea3">
<img width="1728" alt="Chrome ES Send money" src="https://github.com/Expensify/App/assets/57314631/466a73e0-4ede-43c7-bd4a-5cd351544ae2">
<img width="1728" alt="Chrome ES Share code" src="https://github.com/Expensify/App/assets/57314631/c66fa7ef-44f7-46ef-991b-6fd1af3a3559">
<img width="1728" alt="Chrome ES Start a chat" src="https://github.com/Expensify/App/assets/57314631/161e4b29-3765-4f39-871e-7a171db58382">

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->
<img width="1215" alt="Desktop Eng Learn more" src="https://github.com/Expensify/App/assets/57314631/4963c04b-e877-4898-9fd1-7c174e341b62">
<img width="1215" alt="Desktop Eng Refer a friend" src="https://github.com/Expensify/App/assets/57314631/6a245a60-e21e-4caf-afbb-c59578619e2e">
<img width="1215" alt="Desktop Eng Request money" src="https://github.com/Expensify/App/assets/57314631/191715b9-98f1-405c-8970-3791cca396c2">
<img width="1215" alt="Desktop Eng Send money" src="https://github.com/Expensify/App/assets/57314631/7db62f09-8e1f-4b2e-9c40-b3214597c428">
<img width="1215" alt="Desktop Eng Share code" src="https://github.com/Expensify/App/assets/57314631/0d7543c3-aee2-411f-96f9-9c7e0d370b10">
<img width="1215" alt="Desktop Eng Start a chat" src="https://github.com/Expensify/App/assets/57314631/e718a2d8-96aa-461e-8113-2f89cd3ea98a">
<img width="1215" alt="Desktop ES Learn more" src="https://github.com/Expensify/App/assets/57314631/e203135b-1c75-417e-8fee-e6d4b02c54ef">
<img width="1215" alt="Desktop ES Refer a friend" src="https://github.com/Expensify/App/assets/57314631/1acac821-4e59-48bf-a9be-a325691ba388">
<img width="1215" alt="Desktop ES Request money" src="https://github.com/Expensify/App/assets/57314631/93a96e2d-bb9e-41dc-ad1c-c6e8677d0d68">
<img width="1215" alt="Desktop ES Send money" src="https://github.com/Expensify/App/assets/57314631/4adf555b-c659-45af-afae-6f3e4a1f02cc">
<img width="1215" alt="Desktop ES Share code" src="https://github.com/Expensify/App/assets/57314631/fe49fd8e-0650-478c-98b1-d9212602d31e">
<img width="1215" alt="Desktop ES Start a chat" src="https://github.com/Expensify/App/assets/57314631/ca5d14cd-629c-4032-ad3f-faabf36f8345">

</details>

rezkiy37 commented:
@jamesdeanexpensify, please verify that everything looks good and content is correct. 

<details><summary>Share code</summary>

<img width="380" alt="Screenshot 2023-12-02 at 15 43 41" src="https://github.com/Expensify/App/assets/57314631/37cf4f8e-7995-4cc5-aaba-1b5f1b3dc820">

</details> 

<details><summary>Refer a friend</summary>

<img width="380" alt="Screenshot 2023-12-02 at 15 43 44" src="https://github.com/Expensify/App/assets/57314631/6614a860-e7e9-4568-a70d-980ff6994ad7">

</details> 

<details><summary>Request money</summary>

<img width="380" alt="Screenshot 2023-12-02 at 15 43 47" src="https://github.com/Expensify/App/assets/57314631/fa93e1f0-2782-4b5e-98e8-61487e81ebc9">

</details> 


<details><summary>Start a chat</summary>

<img width="380" alt="Screenshot 2023-12-02 at 15 43 50" src="https://github.com/Expensify/App/assets/57314631/645d68db-9680-43fa-b4ec-6d2c4299635d">

</details> 

<details><summary>Send money</summary>

<img width="380" alt="Screenshot 2023-12-02 at 15 43 52" src="https://github.com/Expensify/App/assets/57314631/01cce1e2-5559-4115-87fd-2d1f88019ea2">

</details>

jamesdeanexpensify commented:
I'm going to share these internally today for any final feedback in case we're missing anything. Will be back soon, thanks!

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/32187-referral-page-redesign

rezkiy37 committed:
use a proper const

rezkiy37 commented on src/pages/ReferralDetailsPage.js:
```diff
@@ -40,80 +42,62 @@ const defaultProps = {
+            headerContainerStyles={[styles.staticHeaderImage, styles.justifyContentCenter]}
+            backgroundColor={theme.PAGE_THEMES[SCREENS.SETTINGS.WORKSPACES].backgroundColor}
+        >
+            <Text style={[styles.textHeadline, styles.textAlignCenter, styles.mb3, styles.mt8, styles.ph4]}>{contentHeader}</Text>
```
So, it is already has bottom margins. Why do we need paddings?

rezkiy37 replied:
Done - 3870b4d.

<img width="590" alt="Screenshot 2023-12-07 at 19 37 29" src="https://github.com/Expensify/App/assets/57314631/fd17642f-bd81-4b13-8acd-48d254e30f57">

rezkiy37 commented on src/pages/ReferralDetailsPage.js:
```diff
@@ -40,80 +42,62 @@ const defaultProps = {
-                    enterKeyEventListenerPriority={1}
+            }
+            headerContainerStyles={[styles.staticHeaderImage, styles.justifyContentCenter]}
+            backgroundColor={theme.PAGE_THEMES[SCREENS.SETTINGS.WORKSPACES].backgroundColor}
```
Done - 0d31230.

rezkiy37 committed:
align text left

rezkiy37 committed:
align icon

rezkiy37 commented:
Fixed UI bugs.

<img width="425" alt="Screenshot 2023-12-07 at 19 21 52" src="https://github.com/Expensify/App/assets/57314631/b6d5dfac-60a9-4bd9-8267-47cc159c4e28">

rezkiy37 committed:
add bottom margin for last element

jamesdeanexpensify commented:
We're still making the button "Copy invite link" right? Thanks!

grgia commented:
@parasharrajat it should be 32px

rezkiy37 committed:
translate referral link button

rezkiy37 committed:
decrease spacing

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/32187-referral-page-redesign

rezkiy37 commented:
1. Changed title of the copy button.
2. Updated spacing between the top block and content.

<img width="410" alt="Screenshot 2023-12-08 at 10 56 28" src="https://github.com/Expensify/App/assets/57314631/09492546-7e0a-4e0c-b6a1-cb7a752263fb">

<img width="406" alt="Screenshot 2023-12-08 at 10 57 55" src="https://github.com/Expensify/App/assets/57314631/a4f1a68b-eda2-465e-bbc2-26630849cbcd">

parasharrajat reviewed (approved):
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

:ribbon: :eyes: :ribbon:  C+ reviewed

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/32187-referral-page-redesign

rezkiy37 commented on src/languages/es.ts:
```diff
@@ -2442,31 +2442,30 @@ export default {
+            header: `Invita a un amigo y obtiene $${CONST.REFERRAL_PROGRAM.REVENUE}`,
+            body: `Sé el primero en invitar a un amigo (o a cualquier otra persona) a Expensify y obtiene $${CONST.REFERRAL_PROGRAM.REVENUE} si se convierte en cliente de Expensify. Comparte tu enlace de invitación por SMS, email o publícalo en las redes sociales.`,
         },
         copyReferralLink: 'Copiar enlace de invitación',
```
@parasharrajat, yes, it is.

hayata-suenaga commented on src/pages/ReferralDetailsPage.js:
```diff
@@ -40,80 +42,62 @@ const defaultProps = {
-        </ScreenWrapper>
+            )}
+
+            <MenuItem
```
yep I agree with @rezkiy37 👍

hayata-suenaga reviewed (dismissed):
LGTM

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/32187-referral-page-redesign

rezkiy37 commented:
The conflicts were resolved.

hayata-suenaga reviewed (dismissed):
good work! LGTM!

hayata-suenaga commented:
there are conflicts 😢

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/32187-referral-page-redesign

rezkiy37 commented:
@hayata-suenaga, resolved.

rezkiy37 commented:
A bug was found. Investigating...
<img width="1473" alt="Screenshot 2023-12-12 at 10 36 53" src="https://github.com/Expensify/App/assets/57314631/49283311-0bfd-4f3c-8e9c-4daeeac1b11a">

rezkiy37 committed:
fix merge changes

rezkiy37 commented:
Fixed.
<img width="1716" alt="Screenshot 2023-12-12 at 10 40 19" src="https://github.com/Expensify/App/assets/57314631/1f6c6b14-ced0-4445-b8b2-2c890c10b832">

hayata-suenaga reviewed (approved):
thanks for fixing the merge conflicts 👍

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/7194238476) to staging by https://github.com/hayata-suenaga in version: 1.4.12-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|cancelled 🔪
🍎 iOS 🍎|cancelled 🔪
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/7194238476) to staging by https://github.com/hayata-suenaga in version: 1.4.12-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|cancelled 🔪
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/7194238476) to staging by https://github.com/hayata-suenaga in version: 1.4.12-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/7210852007) to production by https://github.com/Julesssss in version: 1.4.12-2 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

jamesdeanexpensify commented:
Actually, just kidding. It works! Sorry bout that.
